presto-main/src/main/java/com/facebook/presto/sql/analyzer/RefreshMaterializedViewPredicateAnalyzer.java

```diff
@@ -0,0 +1,166 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.sql.analyzer;
+
+import com.facebook.presto.Session;
+import com.facebook.presto.common.QualifiedObjectName;
+import com.facebook.presto.metadata.Metadata;
+import com.facebook.presto.spi.ConnectorMaterializedViewDefinition;
+import com.facebook.presto.spi.MaterializedViewNotFoundException;
+import com.facebook.presto.spi.SchemaTableName;
+import com.facebook.presto.sql.tree.ComparisonExpression;
+import com.facebook.presto.sql.tree.DefaultTraversalVisitor;
+import com.facebook.presto.sql.tree.DereferenceExpression;
+import com.facebook.presto.sql.tree.Expression;
+import com.facebook.presto.sql.tree.Identifier;
+import com.facebook.presto.sql.tree.Literal;
+import com.facebook.presto.sql.tree.LogicalBinaryExpression;
+import com.facebook.presto.sql.tree.Node;
+import com.google.common.collect.ImmutableMap;
+import com.google.common.collect.ImmutableMultimap;
+
+import javax.annotation.Nullable;
+
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+
+import static com.facebook.presto.metadata.MetadataUtil.toSchemaTableName;
+import static com.facebook.presto.sql.analyzer.SemanticErrorCode.NOT_SUPPORTED;
+import static com.facebook.presto.sql.analyzer.SemanticExceptions.missingAttributeException;
+import static java.util.Objects.requireNonNull;
+
+public class RefreshMaterializedViewPredicateAnalyzer
```

Comment:
@jainxrohit, could you help to check if there is anything we could share with #15910?

Reply:
This visitor only visits the level of WHERE expressions but not the entire MV definition SQL statement. So it's more like ExpressionAnalyzer for "DELETE FROM WHERE" (https://github.com/prestodb/presto/blob/7d9746d2b5e63cbceb2e0c36ac8c75984eefdf3e/presto-main/src/main/java/com/facebook/presto/sql/analyzer/StatementAnalyzer.java#L493).

It's like ExpressionAnalyzer vs StatementAnalyzer. So don't think it could merge into MaterializedViewPlanValidator which is at the statement level. This visitor also does view-to-base-table column substitution.

It makes sense to move to xxxMaterializedViewUtils. However, HiveMaterializedViewUtils is at the individual connector level, while this visitor is at the main level for general syntax. I would add a new xxxMaterializedViewUtils at the main level, or prefer to keep this simple Analyzer.